### PR TITLE
Keep existing desktop/menu/taskbar shortcuts after update

### LINKED ISSUE #926: NSIS: Auto-updating leaves a bad shortcut when a user attaches the icon to the Windows Taskbar
* **Version**: Electron-builder: 8.6.0

* **Target**:  Windows NSIS

## User Story
As a Electron-App user, I would like my pinned Taskbar icon to still work after updating, so I can quickly use my app after updates and reduce time spent pinning the icon to my Taskbar after every update. 

## Current situation:
When you do auto-update, it installs it in a new folder, deletes the old folder, and leaves the pinned Taskbar icon referencing a folder that doesn't exist.

### Example workflow:
```
### User side: 
1. user has electron app on Windows Taskbar.
2. User runs electron app on 2.0.21. It is installed in C:\Program Files\electron-app\2.0.21\electron-app.exe 
3. User boots up the app. 
4. User gets install prompt. Accepts it.
5. The app downloads. 
### Electron-App: 
* Creates a new folder 2.0.22. 
* Installs Electron-app in the  2.0.22. 
* Deletes folder 2.0.21.
* leaves the dock icon referencing a folder and .exe that doesn't exist.

```

## Ideal situation:
When you do an auto-update, the pinned Taskbar icon continues to work. 

## Assumptions: 
Currently Electron-App structure is: Program_Files\Electron-App-Name\Version\Electron-App.exe
Uninstalling is easy because it uninstalls the entire `Version` folder.

## Potential Proposal:
Move the .exe outside of the version folder?
Electron-App Structure: 
* Program_files\Electron-App-Name\Electron-App.exe 
* Program_files\Electron-App-Name\Version\Electron-App files (including uninstall)

Note: The above proposal is just a suggestion, I am open to whatever the best technical implementation is (I trust you know better than me).

References: #884

### PR DESCRIPTION
**Pull request by MariaDima**
Close #926

### CODE REVIEW

**Comment by MariaDima** [comment]
Followed mattllyons suggestion under https://github.com/electron-userland/electron-builder/issues/926#issuecomment-271135087.

The check to preserve the shortcuts is based on the application's executable existence.

File: packages/electron-builder/templates/nsis/installSection.nsh [hunk 1]
```diff
@@ -84,6 +84,15 @@ ${IfNot} ${Silent}
   SetDetailsPrint none
 ${endif}
 
+StrCpy $appExe "$INSTDIR\${APP_EXECUTABLE_FILENAME}"
+Var /GLOBAL shortcuts
+StrCpy $shortcuts "--recreate-shortcuts"
+!ifndef allowToChangeInstallationDirectory
+  ${If} ${FileExists} `$appExe`
```
Comment (develar) [inline packages/electron-builder/templates/nsis/installSection.nsh]: Why double quotes not used here?

**Comment by develar** [comment]
You are hero, will be merged soon.

File: packages/electron-builder/templates/nsis/installSection.nsh (hunk 1 above)
Comment (MariaDima) [inline packages/electron-builder/templates/nsis/installSection.nsh]: My bad cleanup!! No evaluation is expected here. I'll update, thanks!

**Pull request closed without merging**